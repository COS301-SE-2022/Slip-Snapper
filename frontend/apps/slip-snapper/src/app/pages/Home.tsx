import {
  IonContent,
  IonHeader,
  IonPage,
  IonTitle,
  IonToolbar,
  IonButtons,
  IonCard,
  IonCardHeader,
  IonCardSubtitle,
  IonCardTitle,
  IonItem,
  IonButton
} from '@ionic/react';
import React from 'react';
import '../theme/variables.css'
import TakePictureButton from '../components/TakePictureButton';
import { NavButtons } from '../components/NavButtons';

const Home: React.FC = () => {
  return (
    <IonPage>
      <IonHeader>
        <IonToolbar color="primary">
          <IonTitle>Slip Snapper</IonTitle>
          <IonButtons slot="end">
            <NavButtons/>
          </IonButtons>
        </IonToolbar>
      </IonHeader>
      <IonContent fullscreen>
      <IonTitle>Recent Reports</IonTitle>
      <IonCard color="primary">
          <IonCardHeader>
            <IonCardSubtitle>10/05/22</IonCardSubtitle>
            <IonCardTitle>Report #11</IonCardTitle>
          </IonCardHeader>

          <IonItem>
            <IonButton fill="outline" slot="end" color="secondary">View</IonButton>
          </IonItem>
        </IonCard>

        <IonCard color="primary">
          <IonCardHeader>
            <IonCardSubtitle>10/05/22</IonCardSubtitle>
            <IonCardTitle>Report #10</IonCardTitle>
          </IonCardHeader>

          <IonItem>
            <IonButton fill="outline" slot="end" color="secondary">View</IonButton>
          </IonItem>
        </IonCard>

        <IonCard color="primary">
          <IonCardHeader>
            <IonCardSubtitle>10/05/22</IonCardSubtitle>
            <IonCardTitle>Report #9</IonCardTitle>
          </IonCardHeader>

          <IonItem>
            <IonButton fill="outline" slot="end" color="secondary">View</IonButton>
          </IonItem>
        </IonCard>

        <TakePictureButton />
      </IonContent>
    </IonPage>
  );
};

export default Home;
