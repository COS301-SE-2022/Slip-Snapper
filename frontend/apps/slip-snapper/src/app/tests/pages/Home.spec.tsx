import React from 'react';
import { fireEvent, render, cleanup } from '@testing-library/react';
import Home from '../../pages/Home';
import renderer from 'react-test-renderer';

jest.mock('@capacitor/filesystem');
jest.mock('../components/TakePictureButton');
jest.mock('../components/NavButtons');
jest.mock('../components/ReportItem');
jest.mock('../components/Graph');
jest.mock('ionicons/icons');
jest.mock("../../api/apiCall");
jest.mock("../../api/Session");
jest.mock('chart.js');
jest.mock('@ionic-native/file-opener/index', () => jest.fn());
beforeEach(cleanup);
describe('Home', () => {
  beforeAll(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  const renderTree = (tree: JSX.Element) => renderer.create(tree);
  it('should render component', () => {
    expect(renderTree(<Home 
    />).toJSON()).toMatchSnapshot();
  });
});
