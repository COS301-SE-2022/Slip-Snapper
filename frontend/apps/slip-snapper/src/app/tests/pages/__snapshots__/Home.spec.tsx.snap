// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render component 1`] = `
<div
  className="ion-page"
>
  <ion-header>
    <ion-toolbar
      color="primary"
    >
      <ion-title>
        Slip Snapper
      </ion-title>
      <ion-buttons
        slot="end"
      >
        <div>
          <ion-button
            onClick={[Function]}
          >
            Home
          </ion-button>
          <ion-button
            onClick={[Function]}
          >
            Reports
          </ion-button>
          <ion-button
            onClick={[Function]}
          >
            Profile
          </ion-button>
          <ion-button
            onClick={[Function]}
          >
            Receipts
          </ion-button>
        </div>
      </ion-buttons>
    </ion-toolbar>
  </ion-header>
  <ion-content
    fullscreen={true}
  >
    <ion-item
      onClick={[Function]}
    >
      <ion-title>
        Recent Reports
      </ion-title>
    </ion-item>
    <div
      className="error-wrapper"
    >
      <ion-card-header>
        <ion-icon
          size="large"
          src="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Alert Circle</title><path d='M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z' stroke-miterlimit='10' class='ionicon-fill-none ionicon-stroke-width'/><path d='M250.26 166.05L256 288l5.73-121.95a5.74 5.74 0 00-5.79-6h0a5.74 5.74 0 00-5.68 6z' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M256 367.91a20 20 0 1120-20 20 20 0 01-20 20z'/></svg>"
        />
        <ion-card-title>
          You currently have no recent reports.
        </ion-card-title>
      </ion-card-header>
    </div>
    <ion-item
      onClick={[Function]}
    >
      <ion-title>
        Expense Summary
      </ion-title>
    </ion-item>
    <div
      className="expense-wrapper"
    >
      <ion-card
        color="primary"
        onClick={[Function]}
      >
        <ion-card-header>
          <ion-card-title>
            Today's Expenditure:
          </ion-card-title>
        </ion-card-header>
        <ion-item
          color="tertiary"
          onClick={[Function]}
        >
          Items Bought: 
          0
        </ion-item>
        <ion-item
          color="tertiary"
          lines="none"
          onClick={[Function]}
        >
          Total Expenditure: R 
          0.00
        </ion-item>
      </ion-card>
      <ion-card
        color="primary"
        onClick={[Function]}
      >
        <ion-card-header>
          <ion-card-title>
            This Week's Expenditure:
          </ion-card-title>
        </ion-card-header>
        <ion-item
          color="tertiary"
          onClick={[Function]}
        >
          Items Bought: 
          0
        </ion-item>
        <ion-item
          color="tertiary"
          lines="none"
          onClick={[Function]}
        >
          Total Expenditure: R 
          0.00
        </ion-item>
      </ion-card>
    </div>
    <ion-item
      onClick={[Function]}
    >
      <ion-title>
        Frequent Purchase Analysis
      </ion-title>
    </ion-item>
    <div
      className="error-wrapper"
    >
      <ion-card-header>
        <ion-icon
          size="large"
          src="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Alert Circle</title><path d='M448 256c0-106-86-192-192-192S64 150 64 256s86 192 192 192 192-86 192-192z' stroke-miterlimit='10' class='ionicon-fill-none ionicon-stroke-width'/><path d='M250.26 166.05L256 288l5.73-121.95a5.74 5.74 0 00-5.79-6h0a5.74 5.74 0 00-5.68 6z' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><path d='M256 367.91a20 20 0 1120-20 20 20 0 01-20 20z'/></svg>"
        />
        <ion-card-title>
          There is currently no data for graphs to be generated.
        </ion-card-title>
      </ion-card-header>
    </div>
    <div
      className="footer-home"
    />
  </ion-content>
  <ion-footer>
    <ion-content>
      <ion-fab
        horizontal="center"
        slot="fixed"
        vertical="bottom"
      >
        <ion-fab-button
          color="secondary"
          onClick={[Function]}
        >
          <ion-icon
            icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Receipt</title><path d='M483.82 32.45a16.28 16.28 0 00-11.23 1.37L448 46.1l-24.8-12.4a16 16 0 00-14.31 0l-25.11 12.41L359 33.7a16 16 0 00-14.36 0L320 46.07l-24.45-12.34a16 16 0 00-14.35-.06L256 46.12l-24.8-12.43a16.05 16.05 0 00-14.33 0L192 46.1l-24.84-12.41a16 16 0 00-19.36 3.94 16.25 16.25 0 00-3.8 10.65V288l.05.05H336a32 32 0 0132 32V424c0 30.93 33.07 56 64 56h12a52 52 0 0052-52V48a16 16 0 00-12.18-15.55zM416 240H288.5c-8.64 0-16.1-6.64-16.48-15.28A16 16 0 01288 208h127.5c8.64 0 16.1 6.64 16.48 15.28A16 16 0 01416 240zm0-80H224.5c-8.64 0-16.1-6.64-16.48-15.28A16 16 0 01224 128h191.5c8.64 0 16.1 6.64 16.48 15.28A16 16 0 01416 160z'/><path d='M336 424v-88a16 16 0 00-16-16H48a32.1 32.1 0 00-32 32.05c0 50.55 5.78 71.57 14.46 87.57C45.19 466.79 71.86 480 112 480h245.68a4 4 0 002.85-6.81C351.07 463.7 336 451 336 424z'/></svg>"
          />
        </ion-fab-button>
        <ion-fab-list
          side="top"
        >
          <ion-fab-button
            onClick={[Function]}
          >
            <ion-icon
              icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>Camera</title><circle cx='256' cy='272' r='64'/><path d='M432 144h-59c-3 0-6.72-1.94-9.62-5l-25.94-40.94a15.52 15.52 0 00-1.37-1.85C327.11 85.76 315 80 302 80h-92c-13 0-25.11 5.76-34.07 16.21a15.52 15.52 0 00-1.37 1.85l-25.94 41c-2.22 2.42-5.34 5-8.62 5v-8a16 16 0 00-16-16h-24a16 16 0 00-16 16v8h-4a48.05 48.05 0 00-48 48V384a48.05 48.05 0 0048 48h352a48.05 48.05 0 0048-48V192a48.05 48.05 0 00-48-48zM256 368a96 96 0 1196-96 96.11 96.11 0 01-96 96z'/></svg>"
            />
          </ion-fab-button>
          <ion-fab-button
            href="/addentry"
            onClick={[Function]}
            router-link="/addentry"
          >
            <ion-icon
              icon="data:image/svg+xml;utf8,<svg xmlns='http://www.w3.org/2000/svg' class='ionicon' viewBox='0 0 512 512'><title>List</title><path stroke-linecap='round' stroke-linejoin='round' stroke-width='48' d='M160 144h288M160 256h288M160 368h288' class='ionicon-fill-none'/><circle cx='80' cy='144' r='16' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><circle cx='80' cy='256' r='16' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/><circle cx='80' cy='368' r='16' stroke-linecap='round' stroke-linejoin='round' class='ionicon-fill-none ionicon-stroke-width'/></svg>"
            />
          </ion-fab-button>
        </ion-fab-list>
      </ion-fab>
    </ion-content>
  </ion-footer>
</div>
`;
